Library/Traceparent.ts

```diff
@@ -0,0 +1,121 @@
+import Util = require("./Util");
+import CorrelationIdManager = require("./CorrelationIdManager");
+
+/**
+ * Helper class to manage parsing and validation of traceparent header. Also handles hierarchical
+ * back-compatibility headers generated from traceparent. W3C traceparent spec is documented at
+ * https://www.w3.org/TR/trace-context/#traceparent-field
+ */
+class Traceparent {
+    private static DEFAULT_TRACE_FLAG = "00";
+    private static DEFAULT_VERSION = "00";
+
+    public legacyRootId: string;
+    public parentId: string;
+    public spanId: string;
+    public traceFlag: string = Traceparent.DEFAULT_TRACE_FLAG;
+    public traceId: string;
+    public version: string = Traceparent.DEFAULT_VERSION;
+
+    constructor(traceparent?: string, parentId?: string) {
+        if (traceparent) { // traceparent constructor
+            // If incoming request contains traceparent: parse it, set operation, parent and telemetry id accordingly. traceparent should be injected into outgoing requests. request-id should be injected in back-compat format |traceId.spanId. so that older SDKs could understand it.
+            if (traceparent.split(",").length > 1) { // If more than 1 traceparent is present, discard both
+                this.traceId = Util.w3cTraceId();
+                this.spanId = Util.w3cTraceId().substr(0, 16);
+            } else {
+                const traceparentArr = traceparent.split("-");
+                const len = traceparentArr.length;
+                if (len >= 4) { // traceparent must contain at least 4 fields
+                    this.version = traceparentArr[0].trim();
```

Comment:
Curious, why do we trim here? to remove heading/trailing spaces?
If it is for getting rid of spaces, might make more sense to move them to header parsing logic versus here.

Reply:
You are right. I've removed these two trims and moved them to before the `split("-")`. I am unsure if I want to move this to the header parsing logic as it is a part of the tracecontext spec and I'd like to have as much of the logic for that in the same place as possible.